RevenueCatUI/View+PresentPaywall.swift

```diff
@@ -480,17 +482,30 @@ private struct PresentingPaywallModifier: ViewModifier {
             }
         }
         .task {
-            guard let info = try? await self.customerInfoFetcher() else { return }
+            await self.updateCustomerInfo()
+        }
+        .onChange(of: scenePhase) { newPhase in
+            // Used when Offer Code Redemption sheet dismisses
```

Comment:
Correct me if I'm wrong but would this be called every time the app becomes active even if it didn't go through the offer code redemption sheet, but something else? I guess it's not a huge deal... but we should be careful about invalidating customer info too often IMO, since it could lead to a user temporarily losing entitlements momentarily. If you can confirm this would only happen on offer code redemption sheet dismissal, I'm ok with this :)

Reply:
Good point from Toni here. The code currently only checks when the `scenePhase` passes to `active`. Perhaps we could restrict this to make it closer to the "when Offer Code Redemption sheet dismisses" event by using https://developer.apple.com/documentation/swiftui/view/onchange(of:initial:_:)-4psgg and only do `updateCustomerInfo()` `if newPhase == .active && oldPhase == .inactive` (this way we avoid calling it when coming from the `.background`)